notebooks/monitoring.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+declare -a TEMP=$(mktemp /temp_monitoring.XXXXXXXX)
+
+function get_disk_info() {
+        # df command and cromwell root field
+        df | grep cromwell_root
+}
+
+function get_disk_usage() {
+        # get disk usage field
+        get_disk_info | awk '{ print $5 }'
+}
+
+function get_mem_info() {
+        # /proc/meminfo
+        cat /proc/meminfo
+}
+
+function get_mem_available() {
+        # mem unused from /proc/meminfo
+        get_mem_info | grep MemAvailable | awk 'BEGIN { FS=" " } ; { print $2 }'
+}
+
+function get_mem_total() {
+        # mem total from /proc/meminfo
+        get_mem_info | grep MemTotal | awk 'BEGIN { FS=" " } ; { print $2 }'
+}
+
+function get_mem_usage() {
+        # memTotal and memAvailable
+        local -r mem_total=$(get_mem_total)
+        local -r mem_available=$(get_mem_available)
+
+        # usage = 100 * mem_used / mem_total
+        local -r mem_used=$(($mem_total-$mem_available))
+        echo "$mem_used" "$mem_total" | awk '{ print 100*($1/$2) }'
+}
+
+function get_cpu_info() {
+        # cpu info from /proc/stat
+        cat /proc/stat | grep "cpu "
+}
+
+function get_cpu_total() {
+        # get the total cpu usage since a given time (including idle and iowait)
+        # user+nice+system+idle+iowait+irq+softirq+steal
+        get_cpu_info | awk 'BEGIN { FS=" " } ; { print $2+$3+$4+$5+$6+$7+$8+$9 }'
+}
+
+function get_cpu_used() {
+        # get the cpu usage since a given time (w/o idle or iowait)
+        # user+nice+system+irq+softirq+steal
+        get_cpu_info | awk 'BEGIN { FS=" " } ; { print $2+$3+$4+$7+$8+$9 }'
+}
+
+function get_cpu_usage() {
+        # get the cpu usage since a given time (w/o idle or iowait)
+        # user+nice+system+irq+softirq+steal
+        local -r cpu_used_cur=$(get_cpu_used)
+
+        # get the total cpu usage since a given time (including idle and iowait)
+        # user+nice+system+idle+iowait+irq+softirq+steal
+        local -r cpu_total_cur=$(get_cpu_total)
+
+        # read in previous cpu usage values
+        read -r -a cpu_prev < ${TEMP}
+        local -r cpu_used_prev=${cpu_prev[0]}
+        local -r cpu_total_prev=${cpu_prev[1]}
+
+        # save current values as prev values for next iteration
+        cpu_prev[0]=$cpu_used_cur
+        cpu_prev[1]=$cpu_total_cur
+        echo "${cpu_prev[@]}" > ${TEMP}
+
+        # usage = 100 * (cpu_used_cur - cpu_used_prev) / (cpu_total_cur-cpu_total_prev)
+        echo "$cpu_used_cur" "$cpu_used_prev" "$cpu_total_cur" "$cpu_total_prev" | awk 'BEGIN {FS=" "} ; { print 100*(($1-$2)/($3-$4)) }'
+
+}
+
+function print_usage() {
+        echo [$(date)]
+        echo \* CPU usage: "$(get_cpu_usage)"%
+        echo \* Memory usage: "$(get_mem_usage)"%
```

Comment:
For consistency, would be nice if `get_cpu_usage` and `get_mem_usage` also appended the `%` to their output.